### PR TITLE
write tests for leaderboard page

### PR DESCRIPTION
**Pull request by MahtabBukhari**
## Issue ticket number and link
- ticket number : #275
- [link](https://github.com/stakwork/sphinx-tribes-frontend/issues/275)

## Evidence 

image

### CODE REVIEW

**Comment by MahtabBukhari** [comment]
@ecurrencyhodler Please review my PR

**Comment by ecurrencyhodler** [comment]
All checks passed. GJ.

File: src/pages/leaderboard/leaderboard.spec.tsx
Comment (kevkevinpal) [inline src/pages/leaderboard/leaderboard.spec.tsx:1]: you need to put the spec file under __tests__ directory

**Comment by kevkevinpal** [comment]
I think the tests look fine if you can put the *.spec.ts file into a directory named __tests__

**Comment by MahtabBukhari** [comment]
@kevkevinpal done

File: src/pages/leaderboard/__tests__/leaderboard.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+  });
+
+  test('clicking on the clip icon for a contributor takes us to their profile', () => {
+    act(async () => {
+      const { getByTestId } = render(<LeaderboardPage />);
+      await waitFor(async () => {
+        const userClipIcon = getByTestId('user-id');
+        fireEvent.click(userClipIcon);
+
+        expect(userClipIcon.getAttribute('href')).toEqual(`/p/${person[0].owner_pubkey}`);
```
[the remark this answers is not in the document]
Comment (MahtabBukhari) [reply src/pages/leaderboard/__tests__/leaderboard.spec.tsx:180]: I think this issue is assigned to umer to fix it

[the remark this answers is not in the document]
Comment (MuhammadUmer44) [reply src/pages/leaderboard/__tests__/leaderboard.spec.tsx:180]: Hi @elraphty, 
This is my issue: you've just merged the unit test. Then, I made some changes, so don't worry.

[the remark this answers is not in the document]
Comment (MuhammadUmer44) [reply src/pages/leaderboard/__tests__/leaderboard.spec.tsx:180]: https://github.com/stakwork/sphinx-tribes-frontend/issues/279

[the remark this answers is not in the document]
Comment (elraphty) [reply src/pages/leaderboard/__tests__/leaderboard.spec.tsx:180]: > I think this issue is assigned to umer to fix it
 
@MahtabBukhari I meant you should change `/p/${person[0].owner_pubkey}` to `/p/${person[0].uuid}``

[the remark this answers is not in the document]
Comment (MuhammadUmer44) [reply src/pages/leaderboard/__tests__/leaderboard.spec.tsx:180]: > > I think this issue is assigned to umer to fix it
> 
> @MahtabBukhari I meant you should change `/p/${person[0].owner_pubkey}` to `/p/${person[0].uuid}`

I've already fixed it. I'm just waiting for this unit test to be merged; then, I'll create my own PR to cover all the necessary changes.

[the remark this answers is not in the document]
Comment (elraphty) [reply src/pages/leaderboard/__tests__/leaderboard.spec.tsx:180]: Oh! okay, merging now

**Pull request merged**